### ISSUE TITLE
Suggestion: Ability to provide a default value

### ISSUE THREAD

**Issue by benyanke** [rossigee/magento-config-envvars#2]
It'd be really useful to provide a default value which is used if the XML env var isn't set.

**Comment by rossigee** [comment]
How/where do you propose the defaults would be set?

**Comment by benyanke** [comment]
I'm new-ish to magento, so take it with a grain of salt.

But as I understand it, you're looking for the $, then parsing that to get the var, right?

Perhaps a syntax like this could work, using double pipes to end the var and start the default.

```
<default_setup>
                <connection>
                    <host>$MAGE_DB_HOST||localhost</host>
                    <username>$MAGE_DB_USER||root</username>
                    <password>$MAGE_DB_PASS||</password>
                    <dbname>$MAGE_DB_NAME||magentodev</dbname>
                    <initStatements><![CDATA[SET NAMES utf8]]></initStatements>
                    <model><![CDATA[mysql4]]></model>
                    <type><![CDATA[pdo_mysql]]></type>
                    <pdoType></pdoType>
                    <active>1</active>
                </connection>
```

**Comment by rossigee** [comment]
That would work :)

**Comment by benyanke** [comment]
I don't have access to a full magento ENV yet, but writing up a basic PR for you to show the idea.

TLDR: If str contains ||, check if env exists, and handle default if env not exists. If no ||, pass $value directly.

**Comment by rossigee** [comment]
Sounds good to me. I look forward to the PR.

**Comment by benyanke** [comment]
I don't have a full magento install available yet, unfortunately, but did some testing on my local server, and it worked for me when I pulled out the code to test as a standalone function. Let me know how it works for you. If I can get this working well, I'll certainly be using it on my company's new magento deploy soon.

**Comment by rossigee** [comment]
I don't either right now. The patch looks good though.

**Comment by rossigee** [comment]
Tagged as 1.0.4.

**Comment by benyanke** [comment]
Sweet - my first non-documentation PR accepted!

**Comment by rvelhote** [comment]
@rossigee I was looking through the changes before deploying and the overriden `xmlentities` method does not return anything if the variable does not start with the dollar sign. The parent `xmlentities` returns a value so we should honor that. Am I missing something?

**Comment by rossigee** [comment]
@rvelhote - it seems to me to already honour the parent's return value - the first (unchanged) line of the function should set $value to the parent's value, then returns that on the last line of the function. So assuming the value doesn't start with a dollar sign, it should be backward-compatible. Or am I missing something? :)

**Comment by rvelhote** [comment]
@rossigee Yes but there is no `return $value` in the last line of the method for the _if the value doesn't start with the dollar sign_ codepath. PhpStorm is also saying that the method is missing a return value.
 
```
public function xmlentities($value = null)
{
        $value = parent::xmlentities($value);

        if(substr($value, 0, 1) == "$") {
            // Removed code for clariry
        }
        
        // It should return $value here otherwise the method will not return anything
}
```

**Comment by benyanke** [comment]
Ah, I see. You are correct. I forgot to add the last else. Submitting new PR to fix this.

**Comment by rossigee** [comment]
Thanks @rvelhote! Well spotted.

Merged and tagged as 1.0.5.
